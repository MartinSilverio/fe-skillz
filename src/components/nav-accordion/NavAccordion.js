import React from 'react';
import NavList from '../navlist/NavList';
import { useHistory, useLocation } from 'react-router-dom';
import {
  Accordion,
  AccordionItem,
  AccordionItemHeading,
  AccordionItemButton,
  AccordionItemPanel,
  AccordionItemState,
} from 'react-accessible-accordion';

import './NavAccordion.scss';

function NavAccordion({ navItems }) {
  const location = useLocation();
  const history = useHistory();
  const navigateTo = (url) => {
    history.push(url);
  };

  return (
    <div className='accordion-container'>
      <Accordion className='nav-accordion' allowZeroExpanded>
        {navItems.map((navItem) => {
          return (
            <AccordionItem
              key={navItem.id}
              uuid={navItem.id}
              className='nav-accordion-item'
            >
              <AccordionItemHeading
                className='nav-accordion-heading'
                onClick={() => {
                  navigateTo(navItem.url);
                }}
              >
                <AccordionItemButton className='nav-accordion-btn'>
                  <span
                    className={`accordion-title ${
                      location.pathname.includes(navItem.url) && 'selected'
                    }`}
                  >
                    {navItem.title}
                  </span>
                  {navItem.children && (
                    <AccordionItemState>
                      {({ expanded }) =>
                        expanded ? (
                          <span className='accordion-icon collapse'>-</span>
                        ) : (
                          <span className='accordion-icon expand'>+</span>
                        )
                      }
                    </AccordionItemState>
                  )}
                </AccordionItemButton>
              </AccordionItemHeading>
              {navItem.children && (
                <AccordionItemPanel className='nav-accordion-panel'>
                  <NavList
                    className='accordion-subnav'
                    navItems={navItem.children}
                  />
                </AccordionItemPanel>
              )}
            </AccordionItem>
          );
        })}
      </Accordion>
    </div>
  );
}

export default NavAccordion;

// {
/* <AccordionItemButton className='nav-accordion-btn'>
                  {navItem.children ? (
                    <span className='accordion-title'>{navItem.title}</span>
                  ) : (
                    <NavLink
                      to={navItem.url}
                      role='menuitem'
                      className='accordion-title'
                      activeClassName='selected'
                      isActive={(match, location) =>
                        location.pathname.includes(navItem.url)
                      }
                    >
                      {navItem.title}
                    </NavLink>
                  )}

                  {navItem.children && (
                    <AccordionItemState>
                      {({ expanded }) =>
                        expanded ? (
                          <span className='accordion-icon collapse'>-</span>
                        ) : (
                          <span className='accordion-icon expand'>+</span>
                        )
                      }
                    </AccordionItemState>
                  )}
                </AccordionItemButton> */
// }
