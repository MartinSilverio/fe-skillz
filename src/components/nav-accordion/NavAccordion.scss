@import '../../colors';

.accordion-container {
  height: 95vh;
  background-color: $mediumGrey;
}

.nav-accordion {
  font-size: 14px;
  text-transform: uppercase;
  .nav-accordion-heading {
    background-color: $mediumGrey;
    border-bottom: 1px solid $beige;

    .nav-accordion-btn {
      .accordion-title {
        font-weight: bold;
      }
      .accordion-icon {
        position: relative;
        line-height: 0.5;
        font-size: 30px;

        &.collapse {
          top: -3px;
        }
      }
      padding: 20px;
      display: flex;
      justify-content: space-between;
    }
  }

  .nav-accordion-panel {
    background-color: $beige;
    .accordion-subnav {
      padding: 0;
      margin: 0;

      .nav-item {
        padding: 20px;
        display: flex;
        justify-content: left;
      }
    }
  }
}
