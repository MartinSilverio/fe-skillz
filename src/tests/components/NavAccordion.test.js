import React from 'react';
import { shallow } from 'enzyme';
import NavAccordion from '../../components/nav-accordion/NavAccordion';

const itemList = [
  {
    title: 'Lorem',
    id: 1,
    selected: false,
    url: '/lorem',
    children: [
      {
        title: 'Dolor',
        id: 11,
        selected: false,
        url: '/lorem/dolor',
      },
      {
        title: 'Ipsum',
        id: 12,
        selected: false,
        url: '/lorem/ipsum',
      },
      {
        title: 'Sit',
        id: 13,
        selected: false,
        url: '/lorem/sit',
      },
    ],
  },
  {
    title: 'Amet',
    id: 2,
    selected: false,
    url: '/amet',
  },
];

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: '/lorem/ipsum',
  }),
}));
describe('NavAccordion', () => {
  it("Should display accessible react accordion with accordion item that have children and accordion items that don't", () => {
    expect.assertions(1);
    expect(shallow(<NavAccordion navItems={itemList} />)).toMatchSnapshot();
  });
});
