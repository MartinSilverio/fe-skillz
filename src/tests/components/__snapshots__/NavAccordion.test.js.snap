// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavAccordion Should display accessible react accordion with accordion item that have children and accordion items that don't 1`] = `
<div
  className="accordion-container"
>
  <Accordion
    allowZeroExpanded={true}
    className="nav-accordion"
  >
    <AccordionItem
      className="nav-accordion-item"
      key="1"
      uuid={1}
    >
      <AccordionItemHeading
        className="nav-accordion-heading"
        onClick={[Function]}
      >
        <AccordionItemButtonWrapper
          className="nav-accordion-btn"
        >
          <span
            className="accordion-title selected"
          >
            Lorem
          </span>
          <AccordionItemState>
            <Component />
          </AccordionItemState>
        </AccordionItemButtonWrapper>
      </AccordionItemHeading>
      <AccordionItemPanel
        className="nav-accordion-panel"
      >
        <NavList
          className="accordion-subnav"
          navItems={
            Array [
              Object {
                "id": 11,
                "selected": false,
                "title": "Dolor",
                "url": "/lorem/dolor",
              },
              Object {
                "id": 12,
                "selected": false,
                "title": "Ipsum",
                "url": "/lorem/ipsum",
              },
              Object {
                "id": 13,
                "selected": false,
                "title": "Sit",
                "url": "/lorem/sit",
              },
            ]
          }
        />
      </AccordionItemPanel>
    </AccordionItem>
    <AccordionItem
      className="nav-accordion-item"
      key="2"
      uuid={2}
    >
      <AccordionItemHeading
        className="nav-accordion-heading"
        onClick={[Function]}
      >
        <AccordionItemButtonWrapper
          className="nav-accordion-btn"
        >
          <span
            className="accordion-title false"
          >
            Amet
          </span>
        </AccordionItemButtonWrapper>
      </AccordionItemHeading>
    </AccordionItem>
  </Accordion>
</div>
`;
